sylvia-derive/src/multitest.rs

```diff
@@ -453,6 +453,48 @@ where
             >
         };
 
+        let instantiate2 = if cfg!(feature = "cosmwasm_1_2") {
+            #[cfg(not(tarpaulin_include))]
+            {
+                quote! {
+                    let msg = #sylvia ::cw_std::to_json_binary(&msg)
+                        .map_err(Into::< #error_type >::into)?;
+                    let sender = #sylvia ::cw_std::Addr::unchecked(sender);
+
+                    let msg = #sylvia ::cw_std::WasmMsg::Instantiate2 {
+                        admin,
+                        code_id: code_id.code_id,
+                        msg,
+                        funds: funds.to_owned(),
+                        label: label.to_owned(),
+                        salt: salt.into(),
+                    };
+                    let app_response = (*code_id.app)
+                        .app_mut()
+                        .execute(sender.clone(), msg.into())
+                        .map_err(|err| err.downcast::< #error_type >().unwrap())?;
+
+                    #sylvia:: cw_utils::parse_instantiate_response_data(app_response.data.unwrap().as_slice())
+                        .map_err(|err| Into::into( #sylvia ::cw_std::StdError::generic_err(err.to_string())))
+                        .map(|data| #proxy_name {
+                            contract_addr: #sylvia ::cw_std::Addr::unchecked(data.contract_address),
+                            app: code_id.app,
+                            _phantom: std::marker::PhantomData::default(),
+                        })
+                }
+            }
+        } else {
+            #[cfg(not(tarpaulin_include))]
+            {
+                quote! {
+                    let err = #sylvia ::cw_std::StdError::generic_err(
+                        "`with_salt` was called, but it requires `cosmwasm_1_2` feature enabled. Consider removing `with_salt` or adding the `cosmwasm_1_2` feature."
```

Comment:
I am not sure about making it an error. If it were possible to emit it as a warning, it would be better - something like "`with_salt` requires `cosmwasm_1_2` to have an effect. The salt would be ignored; consider enabling `cosmwasm_1_2` if you want predictable address initialization support".

Then in the case of missing `cosmwasm_1_2` I would fall back to `instantiate` and ignore salt.

Reply:
My idea was that `with_salt` does nothing if we don't enable the `cosmwasm_1_2` feature.
If called without the feature user most likely want `insantiate2` to be called but it never will.
There is no reason to keep this call in such case and should either be removed or feature should be enabled.

But maybe the error is actually a better solution here.

Reply:
Ok, that technically works as well. I just think its slightly more maintenance added. But I don't really mind about that.